extensions/functions_arithmetic.yaml

```
@@ -637,6 +637,63 @@ aggregate_functions:
         decomposable: MANY
         intermediate: fp64?
         return: fp64?
+  - name: "product"
+    description: Product of a set of values. Returns null for empty input.
+    impls:
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: i8
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64
+        return: i8
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: i16
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64
+        return: i16
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: i32
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64
+        return: i32
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: i64
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64
+        return: i64
```

The comment thread about options went out of date but it still applies.

```suggestion
      - args:
          - name: overflow
            options: [ SILENT, SATURATE, ERROR ]
            required: false
          - value: i8
        nullability: MIRROR
        decomposable: MANY
        intermediate: i64
        return: i8
      - args:
          - name: overflow
            options: [ SILENT, SATURATE, ERROR ]
            required: false
          - value: i16
        nullability: MIRROR
        decomposable: MANY
        intermediate: i64
        return: i16
      - args:
          - name: overflow
            options: [ SILENT, SATURATE, ERROR ]
            required: false
          - value: i32
        nullability: MIRROR
        decomposable: MANY
        intermediate: i64
        return: i32
      - args:
          - name: overflow
            options: [ SILENT, SATURATE, ERROR ]
            required: false
          - value: i64
        nullability: MIRROR
        decomposable: MANY
        intermediate: i64
        return: i64
```